Bug: tenferro-burn is not externally consumable with current dependency resolution

## Summary

`tenferro-burn` is currently not externally consumable with its present dependency specification.

An external consumer resolves Burn to `0.21.0-pre.2`, then fails while compiling `interpol 0.2.1` before any `tenferro-burn` code can run.

## Relevant dependency declaration

- `extension/tenferro-burn/Cargo.toml:16`

```toml
burn = { version = "0.21.0-pre.1", default-features = false, features = ["ndarray", "autodiff"] }
```

## Reproduction

```bash
cargo metadata --manifest-path target/tmp/bug-burn-stub/Cargo.toml --format-version 1 | jq -r '.packages[] | select(.name=="burn" or .name=="burn-core" or .name=="burn-ndarray" or .name=="interpol" or .name=="syn") | "\(.name) \(.version)"'
cargo tree --manifest-path target/tmp/bug-burn-stub/Cargo.toml -p burn | sed -n '1,120p'
cargo check --manifest-path target/tmp/bug-burn-stub/Cargo.toml
```

Resolved versions:

```text
burn 0.21.0-pre.2
burn-core 0.21.0-pre.2
burn-ndarray 0.21.0-pre.2
interpol 0.2.1
syn 1.0.109
syn 2.0.117
```

Observed failure:

```text
error[E0282]: type annotations needed
...
extern location for syn does not exist
```

## Impact

Downstream crates are blocked at build time, so the crate cannot be validated as a consumer-facing integration in its current form.

--- a/extension/tenferro-burn/Cargo.toml
+++ b/extension/tenferro-burn/Cargo.toml
@@ -14,4 +14,4 @@ tenferro-prims = { path = "../../tenferro-prims" }
 tenferro-device = { path = "../../tenferro-device" }
 tenferro-algebra = { path = "../../tenferro-algebra" }
 
-burn = { version = "0.21.0-pre.1", default-features = false, features = ["ndarray", "autodiff"] }
+burn = { version = "=0.21.0-pre.1", default-features = false, features = ["ndarray", "autodiff"] }

